input_buffer.cc

```diff
@@ -337,6 +337,47 @@ input_buffer::consume(const char *str)
 	return false;
 }
 
+bool
+input_buffer::consume_char_literal(unsigned long long &outInt)
+{
+	outInt = (unsigned char)((*this)[0]);
+	cursor++;
+
+	if(outInt != '\\')
+	{
+		return true;
+	}
+	else if(cursor >= size)
+	{
+		return false;
+	}
+
+	outInt = (unsigned char)((*this)[0]);
+	cursor++;
+
+	switch (outInt) {
+		case 'n':
+			outInt = (unsigned char)'\n';
+			break;
+		case 'r':
+			outInt = (unsigned char)'\r';
+			break;
+		case 't':
+			outInt = (unsigned char)'\t';
+			break;
+		case '0':
+			outInt = 0;
+			break;
+		case '\'':
+		case '\\':
+			break;
```

Comment:
Minor style nit, but it’s good to put default first so that nothing accidentally falls through into it.

Reply:
ok. minute